### PR TITLE
feat(net): tag interface index

### PR DESCRIPTION
**Pull request by cemremengu**
### Required for all PRs:

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #

Add interface index as a tag. `ifindex` is a useful alternative for interface names in some particular cases.

### CODE REVIEW

**Comment by cemremengu** [comment]
!signed-cla

File: plugins/inputs/net/net.go [hunk 1]
```diff
@@ -69,6 +69,7 @@ func (n *NetIOStats) Gather(acc telegraf.Accumulator) error {
 
 		tags := map[string]string{
 			"interface": io.Name,
+			"interface_index": fmt.Sprint(interfacesByName[io.Name].Index),
```
Comment (srebhan) [inline plugins/inputs/net/net.go:72]: What is the type of `interfacesByName[io.Name].Index`? Can we please explicitly convert that to string?

**Review by srebhan — COMMENTED** [review]
Thanks for your contribution @cemremengu!
I have a small comment in the code and a more general question. Is there a guarantee on the order of interfaces when returned by `net.Interfaces()` or is it arbitrary? If the order is not deterministic, i.e. same name always has same index, you will add combinatorics to the series output by this plugin. At least I would feel better if this new information is guarded by a (default false) flag to only output this on explicit information on user request. What do you think?

**Comment by cemremengu** [comment]
Hey @srebhan thanks for taking your time to review. The index can change between reboots of the machine so unfortunately it is not persistent.

I am personally using it but most people will probably never need it (given the fact that no one has asked for it until now). Thus,  on a second thought maybe it is better to not include this functionality at all to avoid confusion in general use case.

If you agree I will close the request, however if you think it is still good to include this information with a flag I am happy to contribute!

**Comment by srebhan** [comment]
@cemremengu if you can live without it, I'd say we close the PR. However, if you need it, we can also pursue this further. Your choice.

**Comment by cemremengu** [comment]
Let's close it then. Thank you for taking your time!

**Pull request closed without merging**